lib/game_statistics.rb

```diff
@@ -76,23 +84,21 @@ def average_goals_per_game
     decimal_average.round(2)
   end
 
-  def total_goals_per_season
-    all_games.each do |game|
-      if @total_goals_per_season.include?(game.season)
-        @total_goals_per_season[game.season] += game.away_goals.to_i + game.home_goals.to_i
-      else
-        @total_goals_per_season[game.season] += game.away_goals.to_i + game.home_goals.to_i
-      end
+  def total_goals
+    @all_games.each do |game|
+      @total_goals_per_season[game["season"]] += game["away_goals"].to_i + game["home_goals"].to_i
     end
     @total_goals_per_season
   end
 
-  def average_goals_per_season
-    average_goals_per_season = Hash.new
-    total_goals_per_season.each do |season, goals|
-      average = goals.to_f / count_of_games_by_season[season]
-      average_goals_per_season[season] = average.round(2)
+  def average_goals_by_season
+    count_of_games_by_season
+    total_goals
+    @average_goals_per_season = Hash.new
+    @total_goals_per_season.each do |season, goals|
+      average = goals.to_f / @games_per_season[season]
+      @average_goals_per_season[season] = average.round(2)
     end
-    average_goals_per_season
+    @average_goals_per_season
   end
 end
```

Comment:
Great job refactoring

Reply:
Our team effort for refactor was a lot of fun! I feel like our frustration levels were pretty low today!